chia/server/ws_connection.py

```diff
@@ -362,6 +362,7 @@ async def time_out(req_id, req_timeout):
         await event.wait()
 
         self.pending_requests.pop(message.id)
+        self.pending_timeouts.pop(message.id)
```

Comment:
without fully understanding the details here.. these two calls to `pop()` look like they really belong in a `finally` block

Reply:
also, don't we need to cancel the task and synchronize with it going away before removing the last handle to it?

Reply:
the task itself will finish after the sleep which is 60 seconds, i think that's fine. 
Regarding the pop, yeah I think this code can be improved

Reply:
we have a bunch of error when shutting down about tasks/futures that are not awaited, and attempts to access the database after the connection is closed. I suspect that's because we're not very strict about not "detaching" tasks.

If we shut down before the timeout expires, we'll presumably get another error like that. Perhaps here and now isn't the time to fix that, but I think going forward we should avoid detaching and start cleaning up the ones we have already